### PR TITLE
3.31.0

### PR DESCRIPTION
**Pull request by phansys**

### CODE REVIEW

**Review by VincentLanglet — APPROVED** [review]

**Comment by phansys** [comment]
@VincentLanglet, the issue related to #1337 is here again:
https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/1339/checks?check_run_id=2066847402#step:7:108.

**Comment by VincentLanglet** [comment]
Yes... But I don't understand.

> @VincentLanglet, the issue related to #1337 is here again:
> https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/1339/checks?check_run_id=2066847402#step:7:108.

This is weird, it goes and come back...

Let's revert it then.

**Comment by dmaicher** [comment]
Weird stack trace...

```
it@52f0651d576b:/var/www/SonataDoctrineORMAdminBundle  (3.x) $ SYMFONY_DEPRECATIONS_HELPER='/Sonata\\DoctrineORMAdminBundle\\Filter\\NullFilter/' vendor/bin/simple-phpunit
[critical] Error thrown while running command "doctrine:database:create --no-interaction=1". Message: "An exception occurred in driver: could not find driver"

[critical] Error thrown while running command "doctrine:schema:create". Message: "Schema-Tool failed with Error 'An exception occurred in driver: could not find driver' while executing DDL: CREATE TABLE Category (id VARCHAR(255) NOT NULL, name VARCHAR(255) NOT NULL, PRIMARY KEY(id))"

[critical] Error thrown while running command "doctrine:fixtures:load --no-interaction". Message: "An exception occurred in driver: could not find driver"

PHPUnit 9.4.4 by Sebastian Bergmann and contributors.

Testing 

Self deprecation triggered by Sonata\DoctrineORMAdminBundle\Tests\Admin\FieldDescriptionTest::testOptions:
The "Sonata\DoctrineORMAdminBundle\Filter\NullFilter" class is considered final. It may change without further notice as of its next major version. You should not extend it from "Sonata\DoctrineORMAdminBundle\Filter\EmptyFilter".
Stack trace:
#0 [internal function]: Symfony\Bridge\PhpUnit\DeprecationErrorHandler->handleError(16384, 'The "Sonata\\Doc...', '/var/www/Sonata...', 387, Array)
#1 vendor/symfony/error-handler/DebugClassLoader.php(387): trigger_error('The "Sonata\\Doc...', 16384)
#2 vendor/symfony/error-handler/DebugClassLoader.php(313): Symfony\Component\ErrorHandler\DebugClassLoader->checkClass('Sonata\\Doctrine...')
#3 vendor/symfony/phpunit-bridge/Legacy/SymfonyTestsListenerTrait.php(264): Symfony\Component\ErrorHandler\DebugClassLoader::checkClasses()
#4 vendor/symfony/phpunit-bridge/Legacy/SymfonyTestsListenerForV7.php(59): Symfony\Bridge\PhpUnit\Legacy\SymfonyTestsListenerTrait->endTest(Object(Sonata\DoctrineORMAdminBundle\Tests\Admin\FieldDescriptionTest), 0.003501548)
#5 vendor/bin/.phpunit/phpunit-9.4-0/src/Framework/TestResult.php(449): Symfony\Bridge\PhpUnit\Legacy\SymfonyTestsListenerForV7->endTest(Object(Sonata\DoctrineORMAdminBundle\Tests\Admin\FieldDescriptionTest), 0.003501548)
#6 vendor/bin/.phpunit/phpunit-9.4-0/src/Framework/TestResult.php(979): PHPUnit\Framework\TestResult->endTest(Object(Sonata\DoctrineORMAdminBundle\Tests\Admin\FieldDescriptionTest), 0.003501548)
#7 vendor/bin/.phpunit/phpunit-9.4-0/src/Framework/TestCase.php(884): PHPUnit\Framework\TestResult->run(Object(Sonata\DoctrineORMAdminBundle\Tests\Admin\FieldDescriptionTest))
#8 vendor/bin/.phpunit/phpunit-9.4-0/src/Framework/TestSuite.php(677): PHPUnit\Framework\TestCase->run(Object(PHPUnit\Framework\TestResult))
#9 vendor/bin/.phpunit/phpunit-9.4-0/src/Framework/TestSuite.php(677): PHPUnit\Framework\TestSuite->run(Object(PHPUnit\Framework\TestResult))
#10 vendor/bin/.phpunit/phpunit-9.4-0/src/Framework/TestSuite.php(677): PHPUnit\Framework\TestSuite->run(Object(PHPUnit\Framework\TestResult))
#11 vendor/bin/.phpunit/phpunit-9.4-0/src/TextUI/TestRunner.php(667): PHPUnit\Framework\TestSuite->run(Object(PHPUnit\Framework\TestResult))
#12 vendor/bin/.phpunit/phpunit-9.4-0/src/TextUI/Command.php(148): PHPUnit\TextUI\TestRunner->run(Object(PHPUnit\Framework\TestSuite), Array, Array, true)
#13 vendor/bin/.phpunit/phpunit-9.4-0/src/TextUI/Command.php(101): PHPUnit\TextUI\Command->run(Array, true)
#14 vendor/bin/.phpunit/phpunit-9.4-0/phpunit(22): PHPUnit\TextUI\Command::main()
#15 vendor/symfony/phpunit-bridge/bin/simple-phpunit.php(430): include('/var/www/Sonata...')
#16 vendor/symfony/phpunit-bridge/bin/simple-phpunit(13): require('/var/www/Sonata...')
#17 {main}

**Comment by dmaicher** [comment]
Ok I see. Its somehow caused by `custom_bootstrap.php` that is executed BEFORE the first test.

In there we boot some kernel which probably means its related to the service definition `Sonata\DoctrineORMAdminBundle\Filter\EmptyFilter` somehow? not sure

**Comment by VincentLanglet** [comment]
Indeed:
https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/be010e282956d29a6392f2051f7b06f7feb590d2/tests/custom_bootstrap.php#L19
And:
https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/be010e282956d29a6392f2051f7b06f7feb590d2/tests/App/AppKernel.php#L53

What would be the best solution ? cc @franmomu

**Comment by dmaicher** [comment]
Ah I think it disappears if we mark the service as deprecated in the Symfony DI config

Like

``` 
        <service id="Sonata\DoctrineORMAdminBundle\Filter\EmptyFilter">
            <tag name="sonata.admin.filter.type" alias="doctrine_orm_empty"/>
            <deprecated package="x" version="x" />
        </service>
```

locally it disappears for me then

**Comment by VincentLanglet** [comment]
> Ah I think it disappears if we mark the service as deprecated in the Symfony DI config
> 
> Like
> 
> ```
>         <service id="Sonata\DoctrineORMAdminBundle\Filter\EmptyFilter">
>             <tag name="sonata.admin.filter.type" alias="doctrine_orm_empty"/>
>             <deprecated package="x" version="x" />
>         </service>
> ```
> 
> locally it disappears for me then

Can you try a PR ? :)

**Review by wbloszyk — APPROVED** [review]

**Review by OskarStark — APPROVED** [review]

**Comment by VincentLanglet** [comment]
Can you try to rebase @phansys ?

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,28 @@
 All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
+## [3.31.0](https://github.com/sonata-project/SonataDoctrineORMAdminBundle/compare/3.30.0...3.31.0) - 2021-03-09
```
Comment (VincentLanglet) [inline CHANGELOG.md:5]:
```suggestion
## [3.31.0](https://github.com/sonata-project/SonataDoctrineORMAdminBundle/compare/3.30.0...3.31.0) - 2021-03-11
```

Comment (phansys) [reply CHANGELOG.md]: Done.

**Review by jordisala1991 — APPROVED** [review]

**Review by VincentLanglet — APPROVED** [review]

**Pull request merged**